tripy/tripy/frontend/module/sequential.py

```diff
@@ -0,0 +1,169 @@
+from tripy.frontend.module.parameter import Parameter
+from tripy.frontend.module import Module
+
+
+@export.public_api(document_under="modules/sequential.rst")
+@dataclass
+class Sequential(Module):
+    r"""
+    A module to stack multiple layers or modules in a sequential order. The `Sequential`
+    container can accept either a list of modules or a dictionary of named modules. Modules are
+    added in the order they are passed, and each is called sequentially during the forward pass.
+    """
+
+    modules: Union[Module, Dict[str, Module]]
+    r"""The modules to include in the sequence"""
+
+    def __init__(self, *modules: Union[Module, Dict[str, Module]]) -> None:
+        r"""
+        Args:
+            *modules (Union[Module, Dict[str, Module]]): The modules to include in the sequence.
+                Can be passed as individual positional arguments or as a single dictionary of named modules.
+
+        .. code-block:: python
+            :linenos:
+            :caption: Example
+
+            import tripy as tp
+
+            # Sequential with layers passed as arguments
+            model = tp.Sequential(tp.Linear(1, 3), tp.Linear(3, 2))
+
+            # Sequential with named dictionary of layers
+            model_named = tp.Sequential({'layer1': tp.Linear(1, 3), 'layer2': tp.Linear(3, 2)})
+
+            x = tp.Tensor([1.0])
+            output = model(x)
+        """
+        super().__init__()
+        self.modules = {}
+
+        if len(modules) == 1 and isinstance(modules[0], dict):
+            for name, module in modules[0].items():
+                self.modules[name] = module
+        else:
+            for idx, module in enumerate(modules):
+                self.modules[str(idx)] = module
+
+    def __call__(self, x: "tripy.Tensor") -> "tripy.Tensor":
+        r"""
+        Defines the forward pass by applying each module in the container sequentially to input `x`.
+
+        Args:
+            x (Tensor): The input tensor to pass through the sequence of modules.
+
+        Returns:
+            Tensor: The output tensor after passing through each module in sequence.
+        """
+        for module in self.modules.values():
+            x = module(x)
+        return x
+
+    def __getattr__(self, name) -> Any:
+        """
+        Custom __getattr__ to search both in `modules` dictionary and in other attributes. This is for handling
+        `module = operator.attrgetter(child_name)(module)` calls in tripy/frontend/module/module.py:load_state_dict
+        """
+        # Check if `name` is a key in the modules dictionary
+        if "modules" in self.__dict__ and name in self.modules:
+            return self.modules[name]
+
+        # Fallback to regular attribute access if not found in modules
+        try:
+            return super().__getattr__(name)
+        except AttributeError:
+            raise AttributeError(f"'{self.__class__.__name__}' object has no attribute or module named '{name}'")
```

Comment:
I think we can just let the original attribute error go through here:
```suggestion
        return super().__getattr__(name)
```